infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844Utils.java

```diff
@@ -14,79 +14,77 @@
   }
 
+  public static byte[] flattenBytes(final List<? extends Bytes> bytes) {
+    return flattenBytes(
+        bytes.stream(), bytes.stream().map(Bytes::size).reduce(Integer::sum).orElse(0));
+  }
+
   public static byte[] flattenCommitments(final List<KZGCommitment> commitments) {
     final Stream<Bytes> commitmentsBytes =
         commitments.stream().map(KZGCommitment::getBytesCompressed);
     return flattenBytes(commitmentsBytes, KZGCommitment.KZG_COMMITMENT_SIZE * commitments.size());
   }
 
-  public static String copyTrustedSetupToTempFileIfNeeded(final String trustedSetup)
-      throws IOException {
-    final Optional<Path> maybeFile = getFileOnFileSystemOrLocalClasspath(trustedSetup);
-    if (maybeFile.isPresent()) {
-      return maybeFile.get().toFile().getPath();
-    }
-    final String sanitizedTrustedSetup = UrlSanitizer.sanitizePotentialUrl(trustedSetup);
+  public static TrustedSetup parseTrustedSetupFile(final String filePath) throws IOException {
+    final String sanitizedTrustedSetup = UrlSanitizer.sanitizePotentialUrl(filePath);
     final InputStream resource =
         ResourceLoader.urlOrFile("application/octet-stream")
-            .load(trustedSetup)
-            .orElseThrow(() -> new IllegalStateException(sanitizedTrustedSetup + " is not found"));
-
-    final Path temp = Files.createTempFile("trusted_setup", ".tmp");
-    temp.toFile().deleteOnExit();
-    Files.copy(resource, temp, StandardCopyOption.REPLACE_EXISTING);
-
-    return temp.toFile().getAbsolutePath();
-  }
-
-  private static Optional<Path> getFileOnFileSystemOrLocalClasspath(final String resource) {
-    return getFileOnFileSystem(resource).or(() -> getFileOnLocalClasspath(resource));
-  }
-
-  private static Optional<Path> getFileOnFileSystem(final String resource) {
-    try {
-      return Optional.of(Paths.get(resource)).filter(Files::exists);
-    } catch (final Exception __) {
-      return Optional.empty();
-    }
-  }
+            .load(filePath)
+            .orElseThrow(() -> new FileNotFoundException(sanitizedTrustedSetup + " is not found"));
+    try (BufferedReader reader =
+        new BufferedReader(new InputStreamReader(resource, StandardCharsets.UTF_8))) {
+      // txt format :
+      // Number of G1 points
+      final int g1Size = Integer.parseInt(reader.readLine());
+      // Number of G2 points
+      final int g2Size = Integer.parseInt(reader.readLine());
+      // List of G1 points, one on each new line
+      final List<Bytes48> g1Points = new ArrayList<>();
+      for (int i = 0; i < g1Size; i++) {
+        final Bytes48 g1Point = Bytes48.fromHexString(reader.readLine());
+        g1Points.add(g1Point);
+      }
+      // List of G2 points, one on each new line
+      final List<Bytes> g2Points = new ArrayList<>();
+      for (int i = 0; i < g2Size; i++) {
+        final Bytes g2Point = Bytes.fromHexString(reader.readLine());
+        if (g2Point.size() != 96) {
```

Comment:
I'd suggest using the `destinationSize` param and have the same check as for the `g1Points`